build.sbt

```diff
@@ -121,6 +129,8 @@ lazy val dumpEventSchemaCompilerPlugin = (projectMatrix in file("dump-event-sche
         .getOrElse(Seq.empty)
     },
     libraryDependencies ++= Seq(sprayJson, betterFiles, akkaTyped % Test, akkaPersistence % Test),
+    packageBin / publishArtifact := false, //we want to publish fat jar
+    Compile / packageBin / artifactPath := crossTarget.value / "packageBinPlaceholder.jar", //this ensures that normal jar doesn't override fat jar
```

Comment:
Hmmm so what exactly was the problem happening here? was it some kind of race condition? 🤔

Reply:
Yes, it was race condition. The artefact that took longer to create was the one being published. Running `clean` before `publishM2` ensured that the right artefact won the race.

Reply:
Ok, if you could add a mention of the possible race condition to the code so that we're wary of avoiding a regression in the future